File: Africanacity_Backend/Africanacity_Team24(INF370)/models/Inventory/StockTakeItem.cs
using System.ComponentModel.DataAnnotations;

namespace Africanacity_Team24_INF370_.models.Inventory
{
    public class StockTakeItem//ordered menu items
    {
        [Key]
        public int StockTakeItemId { get; set; }
        public int Quantity { get; set; }
        public int Inventory_ItemId { get; set; }
        public Inventory_Item Inventory_Item { get; set; }
        public int StockTake_Id { get; set; }
        public StockTake StockTake { get; set; }

        public virtual ICollection<WriteOffStock> WriteOffs { get; set; }

        public string Description { get; set; } // Add Description property

    }
}

File: Africanacity_Frontend/Administrator/src/app/shared/stocktakeitem.ts
// stocktake-item.model.ts

// export interface StockTakeItem {
//     inventory_ItemId: number;
//     stockTakeQuantity: number;
//   }

  export interface StockTake {
    stockTakeId: number;
    stockQuantity: number;
    stockTakeDate: Date;
    stockTakeItems: StockTakeItem[];
    items: StockTakeItem[];
  }
  
  export interface StockTakeItem {
    quantity: number;
    inventory_ItemId: number;
    stock_TakeId: number;
  }
  

  
  